frontend/aml_dashboard/src/index.js

```diff
@@ -0,0 +1,746 @@
+  const hands = await detector.estimateHands(img);
+  var x = [];
+  if (hands.length > 0) {
+    const finger = hands[0].keypoints[8];
+    const finger3D = hands[0].keypoints3D[8];
+
+    const fingerHand3D = hands[0].keypoints3D[5];
+
+    x = [finger.x, finger.y,
+        finger3D.x - fingerHand3D.x,
+        finger3D.y - fingerHand3D.y,
+        finger3D.z - fingerHand3D.z];
+
+  } else {
+    x = [0.0, 0.0, 0.0, 0.0, 0.0];
+  }
+  return x;
+}
+
+const hand_stream = input.$images
+  .map(async (img) => ({
+    x : await get_features(img),
+    thumbnail : img,
+    y: label.$value.get(),
+})).awaitPromises();
+
+const hand_window = hand_stream.loop((values, x) => {
+  values.push(x)
+  values = values.slice(-10) // Keep up to 10 most recent
+  return { seed: values, value: values }
+}, [], hand_stream);
+
+function calculate_time_features(arr) {
+  var img = arr[0].thumbnail;
+  const lbl = arr[arr.length-1].y;
+  const feats_per_t = 5;
+  const normal_seq_len = 10;
+  const feats = new Array(feats_per_t * normal_seq_len).fill(0.0);
+  for (let t = 0; t < arr.length; t++) {
+    const m = arr[t];
+    if (m.x.length > 0) {
+      img = generateRedBlob(img, Math.floor(m.x[0]), Math.floor(m.x[1]) );
+      for(let f=0; f < m.x.length; f++) {
+        feats[feats_per_t*t + f] = m.x[f];
+      }
+    }
+  }
+  // Draw the path.
+  return {x : feats, thumbnail : img, y: lbl};
+}
+
+const vis_detection = hand_stream.map( async (v) => (generateRedBlob(v.thumbnail,  Math.floor(v.x[0]), Math.floor(v.x[1]))) ).awaitPromises();
+const instanceViewer = imageDisplay(vis_detection);
+
+hand_window
+  .filter(() => capture.$pressed.get())
+  .map(async (arr) => (calculate_time_features(arr)))
+  .awaitPromises()
+  .subscribe(trainingSet.create);
+
+
+// -----------------------------------------------------------
+// TRAINING
+// -----------------------------------------------------------
+
+const b = button('Train');
+b.title = 'Neural Network Training Launcher';
+const classifier = mlpClassifier({ layers: [64, 32], epochs: 20 }).sync(store, 'mlp-dashboard');
+
+b.$click.subscribe(() => classifier.train(trainingSet));
+
+const url_train = "http://localhost:5000/train/";
+const aml_model = { "ready" : false};
+
+const b_train_AML = button('Train');
+b_train_AML.title = 'AML Training Launcher'
+
+const textAMLTrainStatus = text('Not Trained');
+textAMLTrainStatus.title = 'AML Status';
+
+const n_jobs = number(1);
+n_jobs.title = 'Parallel Training (executions)';
+
+const features = {
+  parameters: {
+    'Iterations': new Stream(10, true),
+    'Percentage of data': new Stream(20, true),
+  },
+};
+
+const controls = modelParameters(features);
+controls.title = 'Choose input values per execution';
+
+console.log('Controls:', controls);
+
+b_train_AML.$click.subscribe( async () => {
+  textAMLTrainStatus.$value.set(' <h1>Training...</h1> <br> <img src="https://i.gifer.com/origin/05/05bd96100762b05b616fb2a6e5c223b4_w200.gif">');
+
```

Comment:
Please add an option for when the model can not be trained. 
```suggestion
textAMLTrainStatus.$value.set(' <h1>AML Model could not be trained. Please try again!</h1>');

```

Reply:
Done

Reply:
Addressed in #2